### PR TITLE
[BEAM-168] IntervalBoundedExponentialBackOff: add deprecated copy of old API

### PR DESCRIPTION
**Pull request by dhalperi**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [X] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [X] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [X] Replace "<Jira issue #>" in the title with the actual Jira issue
       number, if there is one.
 - [X] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

This fixes a bug submitting jobs to Google Cloud Dataflow service.

### CODE REVIEW

**Comment by dhalperi** [comment]
R: @dpmills @davorbonaci @bjchambers

File: sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/util/IntervalBoundedExponentialBackOff.java [hunk 1]
```diff
@@ -54,6 +54,11 @@
   private final long initialIntervalMillis;
   private int currentAttempt;
 
+  @Deprecated
```
Comment (bjchambers) [inline sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/util/IntervalBoundedExponentialBackOff.java]: Maybe add a comment about why this is around and deprecated?

Comment (dhalperi) [inline sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/util/IntervalBoundedExponentialBackOff.java]: Done

**Comment by davorbonaci** [comment]
LGTM

**Comment by dhalperi** [comment]
Travis passed; merging now.

**Pull request closed without merging**